### PR TITLE
Fix #40: Implement Share functionality using NSSharingService

### LINKED ISSUE #40: Make Share toolbar button work

### PR DESCRIPTION
**Pull request by bjtitus**
Simple implementation for sharing using NSSharingServicePicker. I just included the current link and no other text.

### CODE REVIEW

File: Evergreen/MainWindow/MainWindowController.swift [hunk 1]
```diff
@@ -69,6 +69,14 @@ class MainWindowController : NSWindowController, NSUserInterfaceValidations {
 		if item.action == #selector(openArticleInBrowser(_:)) {
 			return currentLink != nil
 		}
+        
+        if item.action == #selector(showShareWindow(_:)) {
+            if let link = currentLink {
+                return URL(string: currentLink) != nil
+            } else {
```
Comment (joeblau) [inline Evergreen/MainWindow/MainWindowController.swift]: Right here, you're safely unwrapping the `currentLInk` optional, but then you're never using the safely unwrapped optional.

Also, this check would be a lot simpler written as this.
1. Early termination
2. Safely use unwrapped optional

```swift
    guard let link = currentLink else {
        return false
    }
    return URL(string: link) != nil
```

File: Evergreen/MainWindow/MainWindowController.swift [hunk 2]
```diff
@@ -89,6 +97,13 @@ class MainWindowController : NSWindowController, NSUserInterfaceValidations {
 			openInBrowser(link)
 		}		
 	}
+    
+    @IBAction func showShareWindow(_ sender: AnyObject) {
+        if let link = currentLink, let url = URL(string: link) {
+            let picker = NSSharingServicePicker(items: [url])
+            picker.show(relativeTo: sender.bounds, of: sender as! NSView, preferredEdge: NSRectEdge.minY)
+        }
```
Comment (joeblau) [inline Evergreen/MainWindow/MainWindowController.swift]: `sender` should shouldn't be force unwrapped.  All of the parameters to the picker can be safely unwrapped before being used.

```swift
if let link = currentLink,
    let url = URL(string: link)
    let sender = sender as? NSView {
        
    let picker = NSSharingServicePicker(items: [url])
    picker.show(relativeTo: sender.bounds, of: sender , preferredEdge: NSRectEdge.minY)
}
```

Comment (bjtitus) [inline Evergreen/MainWindow/MainWindowController.swift]: You're absolutely right, I totally missed this somehow, thanks for taking a look! Cleaned this up and switched to using a `guard` as well.

File: Evergreen/MainWindow/MainWindowController.swift (hunk 1 above)
Comment (bjtitus) [inline Evergreen/MainWindow/MainWindowController.swift]: Not sure how this `currentLink` use snuck in. My local copy has `link` being used.

I'm kind of torn on using `guard` when the statement in the scope is only a single line. I ended up changing to `guard` here and also changing to use `switch` for these cases, which I think better separates each selector path.

**Pull request closed without merging**